lang/src/org/partiql/lang/eval/visitors/SelectListItemAliasVisitorTransform.kt

```diff
@@ -0,0 +1,63 @@
+package org.partiql.lang.eval.visitors
+
+import org.partiql.lang.domains.extractSourceLocation
+
+import org.partiql.lang.domains.PartiqlAst
+import org.partiql.lang.eval.extractColumnAlias
+import org.partiql.pig.runtime.SymbolPrimitive
+
+
+/**
+ * Specifies any previously unspecified select list item aliases.
+ *
+ * Turns:
+ *
+ * ```
+ *  SELECT
+ *      foo,
+ *      b.bat,
+ *      baz + 1
+ * FROM bar AS b
+ * ```
+ *
+ * Into:
+ *
+ * ```
+ *  SELECT
+ *      foo AS foo,
+ *      b.bat AS bat,
+ *      baz + 1 AS _3
+ * FROM bar AS b
+ * ```
+ *
+ * If provided with a query with all of the select list aliases are already specified, an exact clone is returned.
```

Comment:
let's open an issue to opt-out when we have a query that already has AS aliases defined.

Reply:
I'm not clear to me who is doing the opting out.  The partiql service?  Or should the `VisitorTransform` implementation itself detect if there is work to be done?  Detecting if there is work to be done for all input queries may be more expensive than just doing the rewrite... However, if the user knows they've supplied a query with all of the `AS` aliases specified then is your suggestion to provide a way for the user to cause this transform to be skipped?

Reply:
Not blocking. I am not sure about your statement "Detecting if there is work to be done for all input queries may be more expensive than just doing the rewrite". Wouldn't it be cheaper to grab all FROM nodes and check that each input has an `AS` name rather than copying trees?

Another idea is to add to our parser the task to keep track of aliases in FROM clauses and pass a flag as metadata on the root node.